## Objective
We will continue the work on the same dataset of the previous checkpoint, but we will go deeper and study the relation between average_montly_hours and number_project.

## instruction
Load the dataset
Compute the Correlation coefficient with R and python and deduce its value
In the second step, we are going to implement a linear regression of average_montly_hours = a*number_project+b
Deduce the final expression of our regression
Don’t forget to implement your work with both R and python

In [5]:
pip install pandas

     -------------------------------------- 10.0/10.0 MB 109.1 kB/s eta 0:00:00
     --------------------------------------- 14.0/14.0 MB 90.1 kB/s eta 0:00:00
     ------------------------------------ 500.7/500.7 kB 164.3 kB/s eta 0:00:00
Note: you may need to restart the kernel to use updated packages.


# importing the csv file

In [45]:
import pandas as pd

HRM = pd.read_csv('HR_comma_sep.csv')
HRM

satisfaction_level  last_evaluation  number_project  \
0                    0.38             0.53               2   
1                    0.80             0.86               5   
2                    0.11             0.88               7   
3                    0.72             0.87               5   
4                    0.37             0.52               2   
...                   ...              ...             ...   
14994                0.40             0.57               2   
14995                0.37             0.48               2   
14996                0.37             0.53               2   
14997                0.11             0.96               6   
14998                0.37             0.52               2   

       average_montly_hours  time_spend_company  Work_accident  left  \
0                       157                   3              0     1   
1                       262                   6              0     1   
2                       272                   4              0     1   
3                       223                   5              0     1   
4                       159                   3              0     1   
...                     ...                 ...            ...   ...   
14994                   151                   3              0     1   
14995                   160                   3              0     1   
14996                   143                   3              0     1   
14997                   280                   4              0     1   
14998                   158                   3              0     1   

       promotion_last_5years    sales  salary  
0                          0    sales     low  
1                          0    sales  medium  
2                          0    sales  medium  
3                          0    sales     low  
4                          0    sales     low  
...                      ...      ...     ...  
14994                      0  support     low  
14995                      0  support     low  
14996                      0  support     low  
14997                      0  support     low  
14998                      0  support     low  

[14999 rows x 10 columns]

In [13]:
HRM.head()

satisfaction_level  last_evaluation  number_project  average_montly_hours  \
0                0.38             0.53               2                   157   
1                0.80             0.86               5                   262   
2                0.11             0.88               7                   272   
3                0.72             0.87               5                   223   
4                0.37             0.52               2                   159   

   time_spend_company  Work_accident  left  promotion_last_5years  sales  \
0                   3              0     1                      0  sales   
1                   6              0     1                      0  sales   
2                   4              0     1                      0  sales   
3                   5              0     1                      0  sales   
4                   3              0     1                      0  sales   

   salary  
0     low  
1  medium  
2  medium  
3     low  
4     low

In [14]:
HRM.describe()

satisfaction_level  last_evaluation  number_project  \
count        14999.000000     14999.000000    14999.000000   
mean             0.612834         0.716102        3.803054   
std              0.248631         0.171169        1.232592   
min              0.090000         0.360000        2.000000   
25%              0.440000         0.560000        3.000000   
50%              0.640000         0.720000        4.000000   
75%              0.820000         0.870000        5.000000   
max              1.000000         1.000000        7.000000   

       average_montly_hours  time_spend_company  Work_accident          left  \
count          14999.000000        14999.000000   14999.000000  14999.000000   
mean             201.050337            3.498233       0.144610      0.238083   
std               49.943099            1.460136       0.351719      0.425924   
min               96.000000            2.000000       0.000000      0.000000   
25%              156.000000            3.000000       0.000000      0.000000   
50%              200.000000            3.000000       0.000000      0.000000   
75%              245.000000            4.000000       0.000000      0.000000   
max              310.000000           10.000000       1.000000      1.000000   

       promotion_last_5years  
count           14999.000000  
mean                0.021268  
std                 0.144281  
min                 0.000000  
25%                 0.000000  
50%                 0.000000  
75%                 0.000000  
max                 1.000000

# Running the correlation between  average_montly_hours and number_project

In [46]:
np.corrcoef(HRM['number_project'], HRM['average_montly_hours'])

array([[1.        , 0.41721063],
       [0.41721063, 1.        ]])

In [26]:
pip install -U scikit-learn scipy matplotlib

Note: you may need to restart the kernel to use updated packages.


# Running the regression model for number_project and average_montly_hours

In [50]:
from sklearn.linear_model import LinearRegression
x = np.array(HRM['number_project']).reshape((-1, 1)) # the feature must be a minimum of 2D array, hence need to reshape
y = list(HRM['average_montly_hours'])

model = LinearRegression().fit(x, y)
r_sq = model.score(x, y) # This is the R^2 the coefficient of determination showing how well the model fits
print('coefficient of determination: ', r_sq)
print('intercept: ', model.intercept_) #This is the intercept of the model. in this case b
print('coef: ', model.coef_) #This is the coefficient of Regression. in this case a




coefficient of determination:  0.17406471345793328
intercept:  136.7602769397482
coef:  [16.90485267]


# Final expression of our regression

  ### R^2 = 0.17406471345793328
  ### average_montly_hours = 16.90485267*number_project+136.7602769397482
  